core/Cargo.toml

```diff
@@ -50,3 +49,6 @@ toml = "0.8.2"
 tracing = { workspace = true }
 tracing-subscriber = { version = "0.3.15", features = ["env-filter"] }
 tracing-appender = "0.2.2"
+
+[dev-dependencies]
+tempfile = "3.8.1"
```

Comment:
Can we move this to the root `Cargo.toml` for the workspace? This way it is easier to maintain a pinned version for all crates who use it